### PR TITLE
Fix staleness overflow: use subtraction instead of addition in stale check

### PR DESCRIPTION
**Pull request by thedavidmeister**
## Summary
`priceTimestamp + timeout` (or `timestamp + timeout` in the LTS path) can overflow `uint256` when `timeout` is very large, causing a bare `Panic(0x11)` instead of a clean `StalePrice` revert. Overflow is especially easy to trigger on the LTS path where `timestamp` is `uint64` (Flare's `getFeedByIdInWei` return type) while `timeout` remains `uint256`.

Replaced the addition-based check with subtraction:
```
// before
if (block.timestamp > priceTimestamp + timeout)

// after
if (block.timestamp > priceTimestamp && block.timestamp - priceTimestamp > timeout)
```

`block.timestamp - priceTimestamp` cannot overflow because any valid FTSO timestamp is ≤ the current block timestamp. A timeout so large that the subtraction form would never trigger is semantically "infinite" (price never stale), which is correct behaviour.

## Files changed
- `src/lib/price/LibFtsoCurrentPriceUsd.sol` — FTSO-v1 path
- `src/lib/lts/LibFtsoV2LTS.sol` — FTSO-v2 LTS path
- `test/src/lib/op/LibOpFtsoCurrentPriceUsd.t.sol` — regression test (`timeout = type(uint256).max`)

## Test plan
- `forge build` — compiles cleanly.
- New `testRunTimestampPlusTimeoutOverflow` fuzz test: restoring the addition-based check causes `Panic(0x11)`, confirming the mutation is killed.

Closes #102
Closes #105

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved stale-price validation to avoid arithmetic overflow when evaluating `timestamp + timeout`.
  - Preserved existing behavior for extremely large (effectively unlimited) timeout values.
  - Ensures price lookups continue to work reliably even near the maximum `uint256` timestamp.

- **Tests**
  - Added coverage for the overflow edge case to confirm USD price retrieval does not revert/panic under maximal timeout conditions.

### CODE REVIEW

**Comment by thedavidmeister** [comment]
🤖 ai:vetter
Reviewed 4993100e5c23ab49f0e7c13e4c3e33fe91d8c8ca: ready — Fix direction correct: subtraction form avoids uint256 overflow on the addition path; regression tes
cost 437 — oracle staleness-check semantics change, two libs plus test

**Comment by thedavidmeister** [comment]
🤖 ai:producer
Producer note: 3d hand-off — src/generated/FlareFtsoWords.pointers.sol conflicts (BYTECODE_HASH and DESCRIBED_BY_META_HASH differ on both sides). The PR intentionally regenerates the pointers file (its staleness fix changes bytecode) and main has also regenerated it since; the correct merged constants require re-running the build prelude against the merged source, which is not producer-resolvable locally. Merge aborted, branch untouched.

**Comment by thedavidmeister** [comment]
🤖 ai:vetter
Reviewed 36b7bc626e227e8fc2b0f8d3e3790fa0a7ee2a8f: ready — closes #102 #105 — re-vet after merge-update head move, diff unchanged: subtraction staleness form correct on both v1+LTS paths, overflow reachable via rainlang timeout input, discriminating overflow test pinned, pointers regen green
cost 437 — oracle staleness semantics, two libs plus test

**Comment by thedavidmeister** [comment]
🤖 ai:producer
Blocked-infra: 3d merge conflict is in src/generated/FlareFtsoWords.pointers.sol (BYTECODE_HASH + DESCRIBED_BY_META_HASH differ on both sides — this PR intentionally regenerates pointers and main has since regenerated them too); correct merged constants require re-running the build prelude, and rain.flare's meta build is currently divergent local-vs-CI (see rain.flare#186 ai:blocked-infra: regenerated meta binaries differ deterministically from CI's), so a locally-regenerated resolution risks pushing wrong pins. Blocked on resolving that meta-toolchain divergence; merge was aborted, branch untouched (prior 3d note).

File: test/src/lib/op/LibOpFtsoCurrentPriceUsd.t.sol [hunk 1]
```diff
@@ -237,6 +238,33 @@ contract LibOpFtsoCurrentPriceUsdTest is FtsoTest {
...
+
+        mockRegistry();
+        mockFtsoRegistry(FTSO, symbol);
+        activateFtso();
+        mockPriceDetails(priceDetails);
+        mockPrice(FTSO, currentPrice);
+
+        // timeout = max → priceTimestamp + timeout overflows; must not panic.
+        LibFtsoCurrentPriceUsd.ftsoCurrentPriceUsd(symbol, type(uint256).max);
+    }
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply test/src/lib/op/LibOpFtsoCurrentPriceUsd.t.sol:269]: 🤖 ai:producer

Accepted — applied in 119f4c0. The test proved only non-panic; it now captures both return values and asserts `price == currentPrice.price` and `decimals == currentPrice.decimals`.

Mutation-validated that the new assertions actually discriminate: swapping the library return to `(decimals, price)` fails the test (`assertion failed: 5 != 1`), restoring `(price, decimals)` passes (256 fuzz runs). The original overflow mutation (addition-based staleness check → `Panic(0x11)`) still holds.
